app/build.gradle

```diff
@@ -21,8 +23,34 @@ android {
         sourceCompatibility JavaVersion.VERSION_1_8
         targetCompatibility JavaVersion.VERSION_1_8
     }
+    signingConfigs {
+        if (keystorePropertiesFile.isFile()) {
+            logger.lifecycle("keystore.properties exists. Sign APK.")
+            def keystoreProperties = new Properties()
+            keystoreProperties.load(new FileInputStream(keystorePropertiesFile))
+            release {
+                keyAlias keystoreProperties['keyAlias']
+                keyPassword keystoreProperties['keyPassword']
+                storeFile rootProject.file(keystoreProperties['storeFile'])
+                storePassword keystoreProperties['storePassword']
+            }
+        }
+        else
+        {
+            logger.lifecycle("keystore.properties not found. Unable to sign APK.")
+        }
+    }
+    applicationVariants.all { variant ->
+        variant.outputs.all {
+            def versionPart = (!keystorePropertiesFile.isFile() || (variant.name == "debug")) ? variant.name : variant.versionName;
+            outputFileName = "SP-${versionPart}.apk"
```

Comment:
If possible I'd prefer  SP-3.0.2-pull525.nnn / SP-debug / SP-release-not-signed
in pseudocode something like...
def versionPart = variant.versionName; //normal
// unusual cases
if (variant.name == "debug")
  versionPart = variant.name;
else if (!keystorePropertiesFile.isFile()) 
  versionPart = variant.name + "-not-signed";

Reply:
Where is the code that adds the pull# and build# to variant.versionName? I dont' find that code added in this PR. :-)

Reply:
@bryanwussow, the part that add the pull# and build# is in TeamCity configuration.

Reply:
@bryanwussow I have implemented what you wanted:
```
            if (variant.name == "release") {
                if (keystorePropertiesFile.isFile()) {
                    versionPart = variant.versionName;
                } else {
                    versionPart = versionPart + "-not-signed";
                }
            }
```